src/render/decorations/IHyprWindowDecoration.hpp

```diff
@@ -39,6 +39,12 @@ class IHyprWindowDecoration {
 
     virtual bool                     allowsInput();
 
+    virtual void                     onBeginDragOnDeco(const Vector2D&);
+
+    virtual bool                     onEndDragOnDeco(CWindow*, const Vector2D&);
+
+    virtual void                     onMouseDown(const Vector2D&, wlr_pointer_button_event*);
```

Comment:
onMouseDownOnDeco. This is not onMouseDown. Can be alternatively clarified by a comment.